### DISCUSSION TITLE
Nuevo protocolo Teltonika

### DISCUSSION THREAD

**Discussion by blubearec** [eusonlito/GPS-Tracker#28] [General]
En primer lugar, felicitaciones por tu aporte, destacamos el tiempo y esfuerzo dedicado a crear este proyecto.
Nosotros somo una empresa de Desarrollo de Software de Ecuador, tenemos actualmente un proyecto de tracking para un cliente del sector basado en traccar. Por casualidad de la vida, nos topamos con tu repo e hicimos un demo, el cual nos funcionó muy bien para dispositivos Synotrack. Sin embargo, no vimos en la documentación cómo o en dónde desarrollar el soporte para un nuevo protocolo, nosotros usamos dispositivos de Teltonika (FMC130) y nos gustaría incluir el soporte para estos dispositivos. Gracias por tu ayuda!.

**Comment by eusonlito** [comment]
Hola! Muchas gracias por vuestro interés.

Hace tiempo empecé la implementación del protocolo de Teltonika, pero es totalmente diferente a cualquier otro, y la comunicación inicial es bidireccional, entonces necesito rehacer totalmente la integración abstracta de protocolos.

A parte había solicitado a Teltonika un dispositivo para las pruebas, y me han respondido que no les interesan los desarrollos alternativos, con lo cual para poder seguir con las pruebas necesitaría un dispositivo, y como los Sinotrack me funcionan sin problema, pues lo he dejado en pausa.

Por cierto, qué os gusta de esta plataforma y qué mejoras/funcionalidades creeis que sería interesante añadir?

Por qué esta plataforma podría ser una opción mejor que traccar para vosotros?

Saludos!
Lito.

**Reply by adarsh23590** [reply]
So does this work with Teltonika or not? because we are using Teltonika in our company. And from where i get the complete guidelines for this project? I mean complete details of each module? like from where i can get GPX file to import in trips, etc

**Reply by eusonlito** [reply]
Yes, it works with Teltonika devices. And there aren't guide, sorry.

**Reply by adarsh23590** [reply]
Have you change Code Structure of Laravel 11? bcz I have not found proper file structure related to routes and controller and same for other folder structure.

**Reply by eusonlito** [reply]
I'm using the module pattern. Check the `app/Domains` folder.

**Reply by adarsh23590** [reply]
can you please share documentation for that so that i can refer it to do some custom changes as per our reuirement

**Reply by eusonlito** [reply]
All available documentation is on [Wiki](https://github.com/eusonlito/GPS-Tracker/wiki/%5BEN%5D-Home).

**Reply by adarsh23590** [reply]
I am asking documentation for Module pattern. i want to make custom code level changes

**Reply by eusonlito** [reply]
The module pattern is simple. The code logic is in `app/Domains` where each domain is related to a database table. Check the code of some of them to see how simple it is to follow. The views are in `resources/views/domains`.

**Reply by adarsh23590** [reply]
ok thanks.

**Reply by adarsh23590** [reply]
I am not able to get trip data from flespi. How can i get that? do I have to do something in flespi? and what changes i have to do to get all telemetry data in this platform

**Reply by adarsh23590** [reply]
I want this tracker to display all details from flespi like all devices as well as its telemetry here so i do not have to open flespi

**Reply by eusonlito** [reply]
Sorry, I don't know how flespi works.

**Reply by adarsh23590** [reply]
ok but how to get teltonica device details. like trip details and all

**Reply by eusonlito** [reply]
GPS Tracker only store location, speed and direction, the platform doesn't store any other telemetry data.

**Reply by adarsh23590** [reply]
ok but how to connect them with this platform. bcz i have not find any configuration for connecting device with platform. we are using Teltonika FMB120 devices. can we have a teams call?

**Reply by eusonlito** [reply]
Please, check this issue https://github.com/eusonlito/GPS-Tracker/issues/63 and wiki about to create a server https://github.com/eusonlito/GPS-Tracker/wiki/%5BEN%5D-Initial-Configuration

**Reply by adarsh23590** [reply]
I am using Teltonika Device and we have setup it on 5027 port. but when i try to start server it shows server is started but when i refresh page it is not display as started. and in laravel log it shows error like 

[2024-06-28T11:58:25+00:00] production.ERROR: App\Domains\Server\Exception\PortBusy: The server could not be started because the 5027 is not free in /var/www/gps/app/Domains/Server/Action/StartPort.php:137

**Reply by eusonlito** [reply]
Are you using the last version?

**Reply by adarsh23590** [reply]
Can we have call on teams or any other platform where we can discuss this? and what is las Version? I am not able to get you

**Reply by eusonlito** [reply]
You can do `php8.3 ./composer deploy` to upgrade to last version. Sorry, I can't do a teams.

**Reply by adarsh23590** [reply]
We have upgraded to php 8.3 now what to do? we are still not able to make connection between GPS Tracker and Teltonica Device. do we have to run above whole command like php8.3 ./composer deploy ????

**Reply by eusonlito** [reply]
Yes, the command `php8.3 ./composer deploy` will upgrade your platform. Are you using Debian/Ubuntu as SO?

**Reply by adarsh23590** [reply]
We are using 

Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal
---------------------------------------------
and we have try to execute below commands but not able to deploy
abc@gps:/var/www/gps$ php 8.3 ./composer deploy
Could not open input file: 8.3
abc@gps:/var/www/gps$ php8.3 ./composer deploy
Could not open input file: ./composer
abc@gps:/var/www/gps$ php8.3 composer deploy
Could not open input file: composer
abc@gps:/var/www/gps$

**Reply by eusonlito** [reply]
`composer` file exists in repository: https://github.com/eusonlito/GPS-Tracker/blob/master/composer are you sure you are in the right folder? Can yo do a `git pull origin master` first?

**Reply by adarsh23590** [reply]
we are able to do composer deploy, but still we are not able to communicate teltonica device. please guide. i want to share my url and its credentials with you please tell me your email so that i can share. we really want to do this asap.

**Reply by eusonlito** [reply]
Okey, send me a mail to lito-at-eordes-dot-com with url and credentials. I will need ssh access to check if all required commands are available on your system.
